Crash on FuzzyMatch

```
Caused by: java.lang.IndexOutOfBoundsException: 
  at java.util.ArrayList.get (ArrayList.java:411)
  at fr.neamar.kiss.utils.FuzzyScore$MatchInfo.getMatchedSequences (FuzzyScore.java:256)
  at fr.neamar.kiss.dataprovider.AppProvider.requestResults (AppProvider.java:157)
  at fr.neamar.kiss.DataHandler.requestResults (DataHandler.java:246)
  at fr.neamar.kiss.searcher.QuerySearcher.doInBackground (QuerySearcher.java:1077)
  at android.os.AsyncTask$2.call (AsyncTask.java:306)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
```

Maybe #899?

--- a/app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java
+++ b/app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java
@@ -144,6 +144,10 @@ public class AppProvider extends Provider<AppPojo> {
     public void requestResults(String query, Searcher searcher) {
         StringNormalizer.Result queryNormalized = StringNormalizer.normalizeWithResult(query, false);
 
+        if (queryNormalized.codePoints.length == 0) {
+            return;
+        }
+
         FuzzyScore fuzzyScore = new FuzzyScore(queryNormalized.codePoints);
         FuzzyScore.MatchInfo matchInfo = new FuzzyScore.MatchInfo();
 

--- a/app/src/main/java/fr/neamar/kiss/dataprovider/ContactsProvider.java
+++ b/app/src/main/java/fr/neamar/kiss/dataprovider/ContactsProvider.java
@@ -50,9 +50,10 @@ public class ContactsProvider extends Provider<ContactsPojo> {
     @Override
     public void requestResults(String query, Searcher searcher) {
         StringNormalizer.Result queryNormalized = StringNormalizer.normalizeWithResult(query, false);
-        // Search people with composed names, e.g "jean-marie"
-        // (not part of the StringNormalizer class, since we want to keep dashes on other providers)
-        queryNormalized = queryNormalized.replaceAll(Character.codePointAt("-", 0), Character.codePointAt(" ", 0));
+
+        if (queryNormalized.codePoints.length == 0) {
+            return;
+        }
 
         FuzzyScore fuzzyScore = new FuzzyScore(queryNormalized.codePoints);
         FuzzyScore.MatchInfo matchInfo = new FuzzyScore.MatchInfo();

--- a/app/src/main/java/fr/neamar/kiss/dataprovider/SettingsProvider.java
+++ b/app/src/main/java/fr/neamar/kiss/dataprovider/SettingsProvider.java
@@ -24,6 +24,10 @@ public class SettingsProvider extends Provider<SettingsPojo> {
     public void requestResults(String query, Searcher searcher) {
         StringNormalizer.Result queryNormalized = StringNormalizer.normalizeWithResult(query, false);
 
+        if (queryNormalized.codePoints.length == 0) {
+            return;
+        }
+
         FuzzyScore fuzzyScore = new FuzzyScore(queryNormalized.codePoints);
         FuzzyScore.MatchInfo matchInfo = new FuzzyScore.MatchInfo();
 

--- a/app/src/main/java/fr/neamar/kiss/dataprovider/ShortcutsProvider.java
+++ b/app/src/main/java/fr/neamar/kiss/dataprovider/ShortcutsProvider.java
@@ -24,6 +24,10 @@ public class ShortcutsProvider extends Provider<ShortcutsPojo> {
     public void requestResults(String query, Searcher searcher) {
         StringNormalizer.Result queryNormalized = StringNormalizer.normalizeWithResult(query, false);
 
+        if (queryNormalized.codePoints.length == 0) {
+            return;
+        }
+
         FuzzyScore fuzzyScore = new FuzzyScore(queryNormalized.codePoints);
         FuzzyScore.MatchInfo matchInfo = new FuzzyScore.MatchInfo();
 

--- a/app/src/main/java/fr/neamar/kiss/utils/FuzzyScore.java
+++ b/app/src/main/java/fr/neamar/kiss/utils/FuzzyScore.java
@@ -252,7 +252,7 @@ public class FuzzyScore {
 
         public List<Pair<Integer, Integer>> getMatchedSequences() {
             // compute pair match indices
-            List<Pair<Integer, Integer>> positions = new ArrayList<>(matchedIndices.size());
+            List<Pair<Integer, Integer>> positions = new ArrayList<>(this.matchedIndices.size());
             int start = this.matchedIndices.get(0);
             int end = start + 1;
             for (int i = 1; i < this.matchedIndices.size(); i += 1) {

